### PR TITLE
Comparison operators

### LINKED ISSUE #124: Comparison operations API
About implementing comparison operations as functions, before I'd like to discuss the API first.
Operations: http://php.net/manual/en/language.operators.comparison.php

Suggestions are:

| PHP Doc Name | API | 
| --- | --- |
| Equal | `equal`, `equals`, `eq` |
| Identical | `identical`, `strict_eq`, `seq` |
| Not equal | `not_equal`, `not_equals`, `neq` |
| Not identical | `not_identical`, `not_strict_eq`, `nseq` |
| Less than | `less_than`, `lt` |
| Greater than | `greater_than`, `gt` |
| Less than or equal to | `less_than_or_eq`, `lte` |
| Greater than or equal to | `greater_than_or_eq`, `gte` |
| Spaceship | `spaceship`, `lteqgt` |

I personally prefer the abbreviations. e.g.:

```php
// using cond/always from #123

$message = cond([
    [lte(16), always('Can drive in USA')],
    [lte(18), always('Can drive and drink in Brazil')],
    [lte(21), always('Can drink in USA')],
    [gte(60), always('Getting old')],
]);

echo $message(18); // ~> Can drive and drink in Brazil
echo $message(60); // ~> Getting old
```

### PR DESCRIPTION
**Pull request by leocavalcante**
`equal` already implemented on #123 
Closes #124

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -56,11 +59,16 @@
...
             "src/Functional/LastIndexOf.php",
+            "src/Functional/LessThan.php",
+            "src/Functional/LessThanOrEqual.php",
+            "src/Functional/LexicographicCompare.php",
             "src/Functional/Map.php",
             "src/Functional/Maximum.php",
             "src/Functional/Memoize.php",
             "src/Functional/Minimum.php",
             "src/Functional/None.php",
+            "src/Functional/NotEqual.php",
```
Comment (lstrojny) [inline composer.json]: Wouldn't it be easier to implement a `not` function instead?

**Comment by lstrojny** [comment]
Looks good!

File: composer.json (hunk 1 above)
Comment (leocavalcante) [inline composer.json]: Indeed! I will

File: src/Functional/Not.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Logical negation of the given $function
+ *
+ * @param callable $function The function to run value against
+ * @return callable A negation version on the given $function
+ */
+function not(callable $function)
+{
+    return function ($value) use ($function) {
+        return ! $function($value);
```
Comment (lstrojny) [inline src/Functional/Not.php]: superfluous space before "!"

Comment (leocavalcante) [inline src/Functional/Not.php]: I really think it makes it more readable :laughing:

**Pull request merged**